filmreel/src/cut.rs

```diff
@@ -1,49 +1,186 @@
+/// The Register's map of [Cut Variables](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-variable)
+type Variables<'a> = HashMap<&'a str, &'a str>;
+
+#[allow(dead_code)] // FIXME
 impl<'a> Register<'a> {
-    pub fn new() -> Self {
-        Default::default()
-    }
-    pub fn insert(&mut self, key: &'a str, val: &'a str) -> Result<Option<&'a str>, &str> {
+    /// Inserts entry into the Register's Cut Variables/
+    ///
+    /// Returns an Err if the key value is does not consist solely of characters, dashes, and underscores.
+    pub fn insert(&mut self, key: &'a str, val: &'a str) -> Result<Option<&'a str>, Error> {
         lazy_static! {
             // Permit only alphachars dashes and underscores for variable names
             static ref KEY_CHECK: Regex = Regex::new(r"^[A-za-z_]+$").unwrap();
         }
         if !KEY_CHECK.is_match(key) {
-            return Err(
-                "Only alphanumeric characters, dashes, and underscores are permitted in cut variable names: [A-za-z_]");
+            return Err(Error::FrameParsef(VAR_NAME_ERR, key.to_string()));
         }
         Ok(self.vars.insert(key, val))
     }
+
+    /// Gets a reference to the string slice value for the given var name.
+    ///
+    /// [Cut Variable](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-variable)
+    pub fn get_kv(&self, key: &str) -> Option<(&&str, &&str)> {
+        self.vars.get_key_value(key)
+    }
+
+    /// An iterator visiting all Cut Variables in arbitrary order.
     pub fn iter(&self) -> std::collections::hash_map::Iter<&str, &str> {
         self.vars.iter()
     }
+
+    /// Returns a boolean indicating whether Register.vars contains a given key.
+    ///
+    /// [Cut Variable](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#cut-variable)
+    pub fn contains_key(&self, key: &str) -> bool {
+        self.vars.contains_key(key)
+    }
+
+    /// Returns a vector of Match enums enums found in the string provided for use in cut
+    /// operations.
+    ///
+    /// [Read Operation](https://github.com/Bestowinc/filmReel/blob/supra_dump/cut.md#read-operation)
+    pub fn read_match(&self, json_string: &String) -> Result<Vec<Match>, Error> {
+        lazy_static! {
+            static ref VAR_MATCH: Regex = Regex::new(
+                r"(?x)
+                (?P<esc_char>\\)?   # escape character
+                (?P<leading_b>\$\{) # leading brace
+                (?P<cut_var>[A-za-z_]+) # Cut Variable
+                (?P<trailing_b>})?  # trailing brace
+                "
+            )
+            .unwrap();
+        }
+
+        let mut matches: Vec<Match> = Vec::new();
+
+        for mat in VAR_MATCH.captures_iter(json_string) {
+            // continue if the leading brace is escaped but strip "\\" from the match
+            if mat.name("esc_char").is_some() {
```

Comment:
I think you could simplify this block with an `if let`:

```suggestion
            if let Some(name) = mat.name("esc_char") {
```

then instead of having to `.expect()` below you can just use `name`